azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/queryer/TextIOMavenPluginQueryer.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.maven.queryer;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.MojoFailureException;
+import org.beryx.textio.TextIO;
+
+import com.microsoft.azure.common.logging.Log;
+
+public class TextIOMavenPluginQueryer extends MavenPluginQueryer {
+    private static final String FOUND_VALID_VALUE = "Found valid value. Skip user input.";
+    private static final String PROMPT_STRING_WITHOUT_DEFAULTVALUE = "Define value for %s";
+    private TextIO textIO;
+
+    public TextIOMavenPluginQueryer(TextIO textIO) {
+        this.textIO = textIO;
+    }
+
+    @Override
+    public String assureInputFromUser(String attribute, String defaultValue, List<String> options, String prompt) {
+        final String initValue = getInitValue(attribute);
+        if (initValue != null && validateInputByOptions(initValue, options)) {
+            Log.info(FOUND_VALID_VALUE);
+            return initValue;
+        }
+        prompt = StringUtils.isEmpty(prompt) ? getPromptString(attribute) : prompt;
+        return textIO.<String>newGenericInputReader(null)
+                .withNumberedPossibleValues(options).withDefaultValue(defaultValue).read(prompt);
+    }
+
+    @Override
+    public String assureInputFromUser(String attribute, String defaultValue, String regex,
```

Comment:
It seems that we could remove `errorMessage` parameter, and the value is never specified in caller

Reply:
Then we need to refactor ConfigMojo, will be present in another PR